lib/widgets/currency_converter_dialog.dart

```diff
@@ -0,0 +1,230 @@
+import 'dart:async';
+
+import 'package:fixnum/fixnum.dart';
+import 'package:breez/bloc/account/account_bloc.dart';
+import 'package:breez/bloc/account/account_model.dart';
+import 'package:breez/bloc/blocs_provider.dart';
+import 'package:breez/bloc/user_profile/currency.dart';
+import 'package:breez/bloc/user_profile/fiat_conversion.dart';
+import 'package:breez/bloc/user_profile/user_profile_bloc.dart';
+import 'package:breez/services/breezlib/data/rpc.pb.dart';
+import 'package:breez/theme_data.dart' as theme;
+import 'package:flutter/material.dart';
+import 'package:flutter/services.dart';
+
+class CurrencyConverterDialog extends StatefulWidget {
+  final Function(String string) _onConvert;
+
+  CurrencyConverterDialog(this._onConvert);
+
+  @override
+  CurrencyConverterDialogState createState() {
+    return new CurrencyConverterDialogState();
+  }
+}
+
+class CurrencyConverterDialogState extends State<CurrencyConverterDialog> with SingleTickerProviderStateMixin {
+  AnimationController _controller;
+  Animation<Color> _colorAnimation;
+
+  AccountBloc _accountBloc;
+  UserProfileBloc _userProfileBloc;
+  StreamSubscription<AccountModel> _accountSubscription;
+  StreamSubscription<Rates> _exchangeRateSubscription;
+
+  Currency _currency = Currency.BTC;
+  FiatConversion _fiatCurrency;
+  List<FiatConversion> _fiatConversionList = List<FiatConversion>();
+  final TextEditingController _fiatAmountController = new TextEditingController();
+  final FocusNode _fiatAmountFocusNode = FocusNode();
+
+  String _amount;
+  double _exchangeRate;
+
+  bool _isInit = false;
+
+  @override
+  void initState() {
+    super.initState();
+    _amount = _currency.format(Int64(0), includeSymbol: false);
+    _fiatAmountController.addListener(_convertCurrency);
+
+    _controller = AnimationController(vsync: this, duration: Duration(milliseconds: 400));
+    _colorAnimation = new ColorTween(
+      begin: Colors.black45,
+      end: theme.BreezColors.blue[500],
+    ).animate(_controller)
+      ..addListener(() {
+        setState(() {});
+      });
+    // Loop back to start and stop
+    _controller.addStatusListener((status) {
+      if (status == AnimationStatus.completed) {
+        _controller.reverse();
+      } else if (status == AnimationStatus.dismissed) {
+        _controller.stop();
+      }
+    });
+  }
+
+  @override
+  void didChangeDependencies() {
+    if (!_isInit) {
+      _accountBloc = AppBlocsProvider.of<AccountBloc>(context);
+      _userProfileBloc = AppBlocsProvider.of<UserProfileBloc>(context);
+      registerListeners();
+      _isInit = true;
+    }
+    super.didChangeDependencies();
+  }
+
+  void registerListeners() {
```

Comment:
I think instead of all these listeners it will be more simplified to use one StreamBuilder for AccountModel inside the build function.